polaris/dataset/_dataset.py

```diff
@@ -384,44 +443,48 @@ def to_json(self, destination: str) -> str:
 
         Args:
             destination: The _directory_ to save the associated data to.
+            if_exists: Action for handling existing files in the Zarr archive. Options are 'raise' to throw
+                an error, 'replace' to overwrite, or 'skip' to proceed without altering the existing files.
 
         Returns:
             The path to the JSON file.
         """
-        fs.mkdir(destination, exist_ok=True)
-        table_path = fs.join(destination, "table.parquet")
-        dataset_path = fs.join(destination, "dataset.json")
-        zarr_archive = fs.join(destination, "data.zarr")
+        dmfs.mkdir(destination, exist_ok=True)
+        table_path = dmfs.join(destination, "table.parquet")
+        dataset_path = dmfs.join(destination, "dataset.json")
+        new_zarr_root_path = dmfs.join(destination, "data.zarr")
 
         # Lu: Avoid serilizing and sending None to hub app.
         serialized = self.model_dump(exclude={"cache_dir"}, exclude_none=True)
         serialized["table"] = table_path
 
         # Copy over Zarr data to the destination
-        if self.zarr_root is not None:
-            dest = zarr.open(zarr_archive, "w")
-            zarr.copy_all(source=self.zarr_root, dest=dest)
+        if self.uses_zarr:
+            with tmp_attribute_change(self, "_warn_about_remote_zarr", False):
```

Comment:
**Question**
Is that context manager needed, if `_warn_about_remote_zarr` will be set to `False`after the first warning? Seems like more complexity for little gain.